source/guides/tooling/typescript-support.md

```diff
@@ -0,0 +1,34 @@
+---
+title: TypeScript Support
+comments: false
+---
+
+Cypress ships with {% url "official type declarations" https://github.com/cypress-io/cypress/tree/develop/cli/types %} for {% url "TypeScript" https://www.typescriptlang.org/ %}.
+
+# Intellisense
+
+In order to get intellisense working, you can try a few approaches:
+
+## Triple-Slash Directives
+
+We have seen success in VS Code with intellisense by simply adding a [triple-slash directive](http://www.typescriptlang.org/docs/handbook/triple-slash-directives.html) to the head of your JavaScript testing file. This is the simplest way to get intellisense working.
+
+```js
+/// <reference types="Cypress" />
+```
+
+{% img /img/guides/typescript-intellisense-with-reference.gif %}
+
+## tsconfig
+
+If your project is written in TypeScript, you may find it more convenient to include Cypress in your `tsconfig.json`.
+
+```json
+{
+  "include": [
+    "integration/*.ts",
+    "support/*.ts",
+    "../node_modules/cypress"
```

Comment:
I think this JSON block should be:

If you write your tests in TypeScript, you may find it more convenient to add a `tsconfig.json` inside your Cypress folder.

```json
{
  "compilerOptions": {
	"strict": true,
    "baseUrl": "../node_modules",
    "target": "es5",
    "lib": ["es5", "dom"],
    "types": ["cypress"],
  },
  "include": [
    "**/*.ts"
  ]
}
```

The `"types"` will tell the TypeScript compiler to only include type definitions from Cypress. This will address instances where the project also uses `@types/chai` or `@types/jquery`. Since chai and jQuery are namespaces (globals), incompatible versions will cause the package manager (yarn or npm) nest and include multiple definitions and cause conflicts.

Reply:
omg, thank you! I was struggling 😅 I think this has helped finally push it to a shippable state.